qiskit_ibm_provider/ibm_backend_service.py

```diff
@@ -362,9 +362,10 @@ def jobs(
                         )
                         continue
                     job_list.append(job)
-                    if len(job_list) == original_limit:
+                    added_jobs += 1
```

Comment:
Remove - it is incorrect to skip only the jobs that were added. You must skip all the jobs that were returned.

Reply:
changes applied